tests/input_set.py

```diff
@@ -0,0 +1,69 @@
+from abc import ABCMeta, abstractmethod
+
+import openmoc
+
+
+class InputSet(object):
+    """An abstract class for defining OpenMOC test geometries."""
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self):
+        self.materials = {}
+        self.geometry = None
+
+    @abstractmethod
+    def create_materials(self):
+        """Instantiate Materials."""
+        return
+
+    @abstractmethod
+    def create_geometry(self):
+        """Instantiate a Geometry."""
+        self.geometry.initializeFlatSourceRegions()
+        return
+
+
+class PinCellInput(InputSet):
+    """A pin cell hoblem from sample-input/pin-cell."""
+
+    def create_materials(self):
+        """Instantiate C5G7 Materials."""
+        self.materials = \
+            openmoc.materialize.load_from_hdf5(filename='c5g7-mgxs.h5',
+                                               directory='../../sample-input/')
+
+    def create_geometry(self):
+        """Instantiate a pin cell Geometry."""
+
+        zcylinder = openmoc.ZCylinder(x=0.0, y=0.0, radius=1.0, name='pin')
+        left = openmoc.XPlane(x=-2.0, name='left')
+        right = openmoc.XPlane(x=2.0, name='right')
+        top = openmoc.YPlane(y=2.0, name='top')
+        bottom = openmoc.YPlane(y=-2.0, name='bottom')
```

Comment:
Can we change these surface names to `xmin`, `xmax`, etc? With the transition to 3D, it can be unclear whether top and bottom refer to y or z.